### ISSUE TITLE
Sending emails on iOS fails

### ISSUE THREAD

**Issue by matthewrice345** [sidlatau/flutter_email_sender#26]
If on an iOS device the default mail client is removed but Gmail is installed nothing happens. 

Steps to reproduce: 
1. Ensure gmail is installed.
2. Remove the mail client from an iPhone. 
3. Use the plugin 

Results: Nothing happens

**Comment by xellDart** [comment]
Same here

**Comment by balajiks-dev** [comment]
Any solution for this issues? 
I'm facing Module 'flutter_email_sender' not found

**Comment by shadyshrif** [comment]
Same error

I have this error when I send an email from form in flutter.
Unhandled Exception: PlatformException(UNAVAILABLE, defualt mail app not available, null)

And Also

Unhandled Exception: PlatformException(not_available, No email clients found!, null)

**Comment by alvindrakes** [comment]
Same error here too.

**Comment by tudor07** [comment]
I have Mail but I still get this error.

**Comment by tudor07** [comment]
@matthewrice345 have you found some workaround or some other library?

**Comment by matthewrice345** [comment]
From what i've been able to learn if the user on iOS does not have a default mail client setup it fails to send an email. My "solution" for now is to wrap the call in a try/catch and show a snackbar if an error happens. I believe the exception does say something about no default mail client being setup. At this point i'm not entirely sure the library can do much or if this is because iOS lacks robust default apps support. (rumor is it's coming though)

**Comment by tudor07** [comment]
Yes, this is true. It's not this lib's fault. I notice that If I don't have an email account configured this library throes PlatformException with code `not_available`, but if I use a `mailto:` link it starts a screen to configure my account. It would be cool if this library did the same thing.

**Comment by StephanCassel** [comment]
Same problem.
I think @tudor07  is pointing in the right direction: the "mailto:-approach".

**Comment by RachelleBoerck** [comment]
I had the same issue. What helped was setting the default email under "contacts". 
I actually had to download the original ios email app first, since I only had gmail on my phone. The default settings didn't register the gmail account properly at first without the original email app.

**Comment by devjeff** [comment]
The mailto-approach didn't work for me either when the default mail app was not configured. A nice workaround that I found is using the [share plugin](https://pub.dev/packages/share) as a fallbackt if the email sending fails.

**Comment by goneall** [comment]
From reading through the issue - it looks like iOS does not really support linking an email app to another iOS app, so there won't be any solution which uses the emailer on the client.  This would include the URL solution above.

If we want to support iOS - we're going to need to implement this without relying an an email client.

I'll research other packages which can collect feedback.

**Comment by rafaellop** [comment]
> Results: Nothing happens

Something happens... and it's an exception which you can eventually catch and show a message that gmail is not supported.

```
    try {
      await FlutterEmailSender.send(email);      
    } catch (e) {
      if ((e as PlatformException).code == 'not_available') {
         print('Gmail not supported');
      }
    }

```

**Comment by manukumarsb7** [comment]
Facing the same issue. Any solution for this issues?

**Comment by Yousafkhanzadaa** [comment]
Any solution?

**Comment by balajiks-dev** [comment]
Its not the solution but its using try catch block to prevent from crashing the app and intimate the user about the issue.

`    try {
      await FlutterEmailSender.send(email);      
    } catch (e) {
      if ((e as PlatformException).code == 'not_available') {
         print('Gmail not supported');
      }
    }
`
